### ISSUE TITLE
IOOBE on get-field with DUV -> struct -> DUV

### ISSUE THREAD

**Issue by jarohen** [xtdb/crux#2425]
```sql
statement ok
INSERT INTO t2(id, data) VALUES(2, {'foo': 4})

query T rowsort
SELECT t2.data.foo FROM t2
----
4
```

* Not strictly the get-field op - `t2.data` yields a vector-wise `:variable` expression, which calls `IIndirectVector/.copyTo`. 
* We'd like to just call `ValueVector/.copyValueSafe`, but this is buggy in DUV, so we have our own special case.
* Unfortunately, for non-DUV composite vectors, we delegate to their `.copyValueSafe`, but if there is a nested DUV the composite vectors then call back to the original `DUV/.copyValueSafe`, which causes the issue.

As far as we can tell:
* `DUV/.copyValueSafe` isn't safe - it doesn't create space in the destination offset buffer for the copied value.
* We can't tell what's supposed to set the destination offset value - as far as we can tell, it'll always be 0.

**Comment by refset** [comment]
Consolidating my test case from xtdb/xtdb#2334 here (assuming that resolving this issue is sufficient to make it work):
```
jdt=> INSERT INTO a (id, obj) VALUES (1234, {'a':123,'b':[1,2]});
INSERT 0 0
jdt=> SELECT * FROM a AS x (id, obj);
  id  |            obj            
------+---------------------------
 1234 | {"a":123,"b":[null,null]}
(1 row)
```

**Comment by jarohen** [comment]
could we confirm that this one is still an issue following the recent writer work? if so, let's talk about prios before diving into a fix

**Comment by FiV0** [comment]
Closed in [14c401d8](https://github.com/xtdb/xtdb/commit/14c401d84baa37f2f79ced47759c69a4a069b8fd)
